x-pack/plugins/observability/public/hooks/slo/use_fetch_active_alerts.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useMemo } from 'react';
+import { HttpSetup } from '@kbn/core/public';
+import { BASE_RAC_ALERTS_API_PATH } from '@kbn/rule-registry-plugin/common';
+
+import { useDataFetcher } from '../use_data_fetcher';
+
+type SloId = string;
+
+interface Params {
+  sloIds: SloId[];
+  sloAlertIndex?: string;
+}
+
+export interface ActiveAlerts {
+  count: number;
+  ruleIds: string[];
+}
+
+type ActiveAlertsMap = Record<SloId, ActiveAlerts>;
+
+export interface UseFetchActiveAlerts {
+  data: ActiveAlertsMap;
+  loading: boolean;
+  error: boolean;
+}
+
+interface FindApiResponse {
+  aggregations: {
+    perSloId: {
+      buckets: Array<{
+        key: string;
+        doc_count: number;
+        perRuleId: { buckets: Array<{ key: string; doc_count: number }> };
+      }>;
+    };
+  };
+}
+
+const EMPTY_ACTIVE_ALERTS_MAP = {};
+
+export function useFetchActiveAlerts({ sloIds = [] }: { sloIds: SloId[] }): UseFetchActiveAlerts {
+  const { data: sloAlertIndex } = useDataFetcher<null, string | undefined>({
+    paramsForApiCall: null,
+    initialDataState: undefined,
+    executeApiCall: fetchAlertIndex,
+    shouldExecuteApiCall: useCallback(() => true, []),
+  });
```

Comment:
💬 This can be simplified (and removed) as soon as this PR is merged https://github.com/elastic/kibana/pull/151567

Reply:
Updated